tests/failpoints/cases/test_async_fetch.rs

```diff
@@ -234,3 +234,38 @@ fn test_node_async_fetch_leader_change() {
         must_get_equal(&cluster.get_engine(1), &k, &v);
     }
 }
+
+#[test]
+fn test_node_gc_entry_cache() {
```

Comment:
I think we can add some somments(conditions and expections) to make this test case more clear. 

Maybe 
```
The entry cache should be kept when an alive peer's last index < cache's first index.
```